types/location_muxer.go

```diff
@@ -0,0 +1,134 @@
+package types
```

Comment:
I am not so sure that such a "fat" type with so much logic should be here, in the `types` module. Maybe leave just an interface here so it can be used in the other types, but move the implementation somewhere else?